src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralServiceTest.kt

```diff
@@ -970,77 +970,134 @@ class ReferralServiceTest @Autowired constructor(
+    assertThat(changeLogReturned.firstOrNull()?.reasonForChange).isNotBlank
+    assertThat(changeLogReturned.firstOrNull()?.topic).isEqualTo(AmendTopic.COMPLETION_DATETIME)
   }
 
   @Test
-  fun `updateReferralDetails updates further information and completion deadline update when referral details doesnt exists`() {
+  fun `updateReferralDetails updates maximum enforceable days`() {
     val authUser = AuthUser("123457", "delius", "bernard.beaks")
     val user = userFactory.create("pp_user_1", "delius")
     val description = ""
     val id = UUID.randomUUID()
-    val completionDate = LocalDate.of(2022, 3, 28)
-    val completionDateToChange = LocalDate.of(2022, 7, 1)
     val intervention = interventionFactory.create(description = description)
-
     val referral = referralFactory.createSent(
       createdAt = OffsetDateTime.now(),
       createdBy = user,
       id = id,
       sentAt = OffsetDateTime.now(),
       serviceUserCRN = "crn",
+      intervention = intervention
+    )
+    val referralDetails = referralDetailsFactory.create(
+      referralId = id,
+      createdAt = OffsetDateTime.now(),
+      createdBy = authUser,
+      id = UUID.randomUUID(),
+      maximumNumberOfEnforceableDays = 15,
+      saved = true
+    )
+    whenever(userMapper.fromToken(jwtAuthentionToken)).thenReturn(authUser)
+
+    val referralToUpdate = UpdateReferralDetailsDTO(20, null, "new information", "we decided 10 days wasn't enough")
+    val referralDetailsReturned = referralService.updateReferralDetails(referral, referralToUpdate, user)
+    val referralDetailsValue = referralService.getReferralDetailsById(referralDetailsReturned?.id)
+    val changeLogReturned = changelogRepository.findAll().filter { x -> x.referralId == id }
+
+    assertThat(referralDetailsValue?.referralId).isEqualTo(referral.id)
+    assertThat(referralToUpdate.furtherInformation).isEqualTo(referralDetailsValue?.furtherInformation)
+    assertThat(changeLogReturned.size).isEqualTo(1)
+    assertThat(referralToUpdate.maximumEnforceableDays).isEqualTo(referralDetailsValue?.maximumEnforceableDays)
+
+    assertThat(changeLogReturned.firstOrNull()?.referralId).isEqualTo(referral.id)
+    assertThat(changeLogReturned.firstOrNull()?.newVal).isNotNull
+    assertThat(changeLogReturned.firstOrNull()?.newVal?.values).isNotEmpty
+    assertThat(changeLogReturned.firstOrNull()?.newVal?.values?.get(0)).isEqualTo("20")
+    assertThat(changeLogReturned.firstOrNull()?.oldVal).isNotNull
+    assertThat(changeLogReturned.firstOrNull()?.oldVal?.values).isNotEmpty
+    assertThat(changeLogReturned.firstOrNull()?.oldVal?.values?.get(0)).isEqualTo(referralDetails.maximumEnforceableDays.toString())
+    assertThat(changeLogReturned.firstOrNull()?.reasonForChange).isNotBlank
+    assertThat(changeLogReturned.firstOrNull()?.topic).isEqualTo(AmendTopic.MAXIMUM_ENFORCEABLE_DAYS)
+  }
+
+  @Test
+  fun `updateReferralDetails updates draft Referrals`() {
```

Comment:
Can this go to the `DraftReferralServiceTest` please?

Reply:
sure will move it there